File: tkpdpdp/src/main/java/com/tokopedia/tkpdpdp/estimasiongkir/presentation/adapter/RatesEstimationServiceAdapter.java
package com.tokopedia.tkpdpdp.estimasiongkir.presentation.adapter;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.tokopedia.tkpdpdp.R;
import com.tokopedia.tkpdpdp.estimasiongkir.ShippingServiceModel;
import com.tokopedia.tkpdpdp.estimasiongkir.presentation.viewholder.RatesEstimationServiceViewHolder;

import java.util.ArrayList;
import java.util.List;

public class RatesEstimationServiceAdapter extends RecyclerView.Adapter<RatesEstimationServiceViewHolder> {
    private List<ShippingServiceModel> shippingServiceModels = new ArrayList<>();

    @NonNull
    @Override
    public RatesEstimationServiceViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new RatesEstimationServiceViewHolder(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_rates_estimation_service, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull RatesEstimationServiceViewHolder holder, int position) {
        holder.bind(shippingServiceModels.get(position));
    }

    @Override
    public int getItemCount() {
        return shippingServiceModels.size();
    }

    public void updateShippingServices(List<ShippingServiceModel> shippingServiceModels){
        this.shippingServiceModels.clear();
        this.shippingServiceModels.addAll(shippingServiceModels);
        notifyDataSetChanged();
    }
}

File: tkpdpdp/src/main/java/com/tokopedia/tkpdpdp/estimasiongkir/presentation/adapter/RatesEstimationServiceAdapter.kt
package com.tokopedia.tkpdpdp.estimasiongkir.presentation.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup

import com.tokopedia.tkpdpdp.R
import com.tokopedia.tkpdpdp.estimasiongkir.data.model.ShippingServiceModel
import com.tokopedia.tkpdpdp.estimasiongkir.presentation.viewholder.RatesEstimationServiceViewHolder

import java.util.ArrayList

class RatesEstimationServiceAdapter : RecyclerView.Adapter<RatesEstimationServiceViewHolder>() {
    private val shippingServiceModels = ArrayList<ShippingServiceModel>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RatesEstimationServiceViewHolder {
        return RatesEstimationServiceViewHolder(LayoutInflater.from(parent.context)
                .inflate(R.layout.item_rates_estimation_service, parent, false))
    }

    override fun onBindViewHolder(holder: RatesEstimationServiceViewHolder, position: Int) {
        holder.bind(shippingServiceModels[position])
    }

    override fun getItemCount() = shippingServiceModels.size

    fun updateShippingServices(shippingServiceModels: List<ShippingServiceModel>) {
        this.shippingServiceModels.clear()
        this.shippingServiceModels.addAll(shippingServiceModels)
        notifyDataSetChanged()
    }
}
